app/services/analytics.rb

```diff
@@ -143,11 +147,12 @@ def sp_request_attributes
   end
 
   def differentiator
-    sp_request_url = session&.dig(:sp, :request_url)
-    return {} if sp_request_url.blank?
+    @differentiator ||= begin
```

Comment:
Since this can be nil, we could return if it's already been defined to avoid running again

```suggestion
    return @differentiator if defined?(@differentiator)
    @differentiator ||= begin
```